Geometry/CaloEventSetup/plugins/CaloTowerConstituentsMapBuilder.cc

```diff
@@ -130,17 +130,16 @@ void CaloTowerConstituentsMapBuilder::assignEEtoHE(const CaloGeometry* geometry,
   //get list of EE detids
   const std::vector<DetId>& vec(geomEE->getValidDetIds());
   //loop over EE detids
-  for(std::vector<DetId>::const_iterator detId_itr =vec.begin();
-      detId_itr != vec.end(); detId_itr++){
+  for(auto detId_itr : vec){
```

Comment:
It looks like `DetId` has user-defined default constructors:
https://github.com/cms-sw/cmssw/blob/master/DataFormats/DetId/interface/DetId.h#L26-L32
as do some of the derived types, e.g.:
https://github.com/cms-sw/cmssw/blob/master/DataFormats/HcalDetId/interface/HcalDetId.h#L33-L42
As I would expect based on that, both `DetId` and `HcalDetId` fail `std::is_pod`.